nupic/research/connections.py

```diff
@@ -23,14 +23,28 @@
 
 
 
+class SynapseData(object):
+
+
+  __slots__ = ("segment", "presynapticCell", "permanence")
+
+
+  def __init__(self, segment, presynapticCell, permanence):
+    self.segment = segment
+    self.presynapticCell = presynapticCell
+    self.permanence = permanence
+
+
+  def __eq__(self, other):
+    return (self.segment, self.presynapticCell, self.permanence) == other
+
+
+
 class Connections(object):
   """
   Class to hold data representing the connectivity of a collection of cells.
   """
 
-  SynapseData = namedtuple("SynapseData", ["segment",
-                                           "presynapticCell",
-                                           "permanence"])
 
```

Comment:
@chetan51 From my limited knowledge, would this affect nupic_studio. Issues revolving around "TypeError: 'SynapseData' object is not iterable' in this line https://github.com/nupic-community/nupic.studio/blob/master/nupic_studio/htm/node_region.py#L472 Be great if you could let me know if it is, and whether thee is an easy fix in nupic_studio for it?